GoogleForCreators/web-stories-wp
# Stabilize Custom Meta Boxes Support

## Summary

Stabilizes the existing meta boxes integration that is currently behind a feature flag, enabling it by default.

Part 1 was done in #5048

## Relevant Technical Choices



## To-do

* [ ] Completely remove feature flag. (follow-up PR)
* [x] Fix issues with author users not being able to save meta boxes.
* [x] Confirm UX.

## User-facing changes

https://user-images.githubusercontent.com/841956/104297181-d80c3780-54c2-11eb-8bd5-825d37f0aaf4.mp4

When a user/site doesn't have any third-party integrations, the editor will look the same as today:

<img width="1680" alt="Screenshot 2021-01-12 at 10 41 11" src="https://user-images.githubusercontent.com/841956/104297071-b27f2e00-54c2-11eb-8114-ce4d735d39ba.png">

## Testing Instructions

1. Activate the `All in One SEO Pack` and `Polylang` plugins
1. Open story editor
1. Verify that two meta boxes are shown at the bottom (like in the screenshots above)
1. Verify that meta boxes can be opened and collapsed
1. Write something in the input fields in the meta box(es) and save the story.
1. Verify that changes have been saved and persist even after page reload.
1. Verify that the "Update" button is _not_ disabled, despite there not being any new changes now

---



Fixes #5519
